webapp/survey/nodeDefs/actions.js

```diff
@@ -47,15 +47,17 @@ const _putNodeDefPropsDebounced = (nodeDef, key, props, propsAdvanced) => deboun
 )
 
 const _updateParentLayout = (nodeDef, deleted = false) => async (dispatch, getState) => {
-  const survey = SurveyState.getSurvey(getState())
+  const state = getState()
+  const survey = SurveyState.getSurvey(state)
+  const surveyCycleKey = SurveyState.getSurveyCycleKey(state)
   const nodeDefParent = Survey.getNodeDefParent(nodeDef)(survey)
 
-  if (NodeDef.isEntity(nodeDefParent) && NodeDefLayout.isRenderTable(nodeDefParent)) {
-    const nodeDefParentLayout = NodeDefLayout.getLayout(nodeDefParent)
+  if (NodeDefLayout.isRenderTable(surveyCycleKey)(nodeDefParent)) {
     const nodeDefUuid = NodeDef.getUuid(nodeDef)
+    const layoutChildren = NodeDefLayout.getLayoutChildren(surveyCycleKey)(nodeDefParent)
 
-    const newLayout = deleted ? R.without([nodeDefUuid])(nodeDefParentLayout) : R.append(nodeDefUuid)(nodeDefParentLayout)
-    dispatch(putNodeDefProp(nodeDefParent, NodeDefLayout.nodeDefLayoutProps.layout, newLayout))
+    const layoutChildrenUpdate = deleted ? R.without([nodeDefUuid])(layoutChildren) : R.append(nodeDefUuid)(layoutChildren)
```

Comment:
layoutChildrenUpdate: it should be "layoutChildrenUpdated"

Reply:
ok